### DISCUSSION TITLE
Quick overview from forking to creating a pull request?

### DISCUSSION THREAD

**Discussion by ldeck** [magit/forge#354] [Ideas]
Hey there — firstly thanks so much for this package!

I've read through the manual and looked at some videos around, but would love to see a step by step guide to using forge to fork a repo, create a feature branch, and submit a pull request. I'd be more than happy to create such a pull request for forge's documentation itself once I'm clear on the choices.

(Let's assume forge has been configured with the necessary creds for GitHub for me 'ldeck').

So here's the steps I think is involved — any suggested clarifications or simplifications would be great to hear.
NB: bold text probably areas to focus on for clarity.

1. clone magit/forge via magit:
  a. M-x magit-clone 
  b. The mini buffer asks: `Clone from [u]rl or name, [p]ath, or [l]ocal url`
  c. I chose `u` 
  d. Clone from url or name: `magit/forge`
  e. Chosen directory: `~/Documents/dev/emacs/magit/forge`
  f. **Set remote.pushDefault to "origin" (y or n): Choose y for now?**
  g. done
2. fork with forge
  a. 'cf (i.e., forge-fork) 
  b. Fork to: ldeck
  c. **Remote name: defaults to 'ldeck' (should I type origin to replace the original origin?)**
  d. magit shows me `81f4191 origin/master ldeck/master Remove stray character literal`
  e. done
3. Create feature
  a. bc (M-x magit-branch-create)
  b. Create branch starting at: master
  c. Name for new branch: guide-to-creating-pull-requests
  d. edit relevant files (presumably forge.org) in this case
  e. commit
  f. **push branch to 'ldeck' -- magit is still defaulting to wanting to push to 'origin', so have to select push to other**
4. Initiate pull-request
  a. todo

**Comment by tarsius** [comment]
:wave: Such a guide would certainly be useful. The manual isn't the best place for it though. I would recommend a blog post, that would cause the most attention in the short run. Later you could add it to the forge wiki.

The steps sound right. For (3) you might want to mention the spin-off variant (`b s`). (4) is basically `' c p RET RET some message RET`, but you probably are aware of that already.

**Reply by ldeck** [reply]
Thanks @tarsius. Are you able to comment on 1f, 2c, and 3f (should it by now default to pushing to ldeck repo instead)?

How do you go about this?

**Reply by tarsius** [reply]
Say no at 1f if you intend to add your own fork later. Use `b p...` to cycle between the possible values for the push-remote if you need to later change it.

**Reply by ldeck** [reply]
Thanks @tarsius.
Should forge-fork ask this question anyway if already set prior to calling magit-remote-add? 
`(y-or-n-p (format "Update `remote.pushDefault' to \"%s\"? " fork))`

Users might typically answer `y` when initially cloning a repo prior to forking it (and there might be some lag between the two activities). So given the usual (I assume) intension of forking a repo is for it to become primary, would it make sense to prompt the user about this?

**Reply by tarsius** [reply]
> Should forge-fork ask this question anyway if already set prior to calling magit-remote-add?

I don't think it is worth the effort, so no.
